src/store/ChartStore.js

```diff
@@ -310,6 +310,80 @@ class ChartStore {
+                    if (!field || highLowBars) field = 'Close';
+                    let prevClose, currentClose;
+                    do {
+                        prevClose = chart[whichSet][--l][field];
+                        currentClose = prevClose;
+                        if (l === 0) break;
+                    } while (currentClose === null);
+                    if (whichSet === 'dataSet' && chart.currentQuote) {
+                        currentClose = chart.currentQuote[field];
+                    } else if (chart[whichSet].length >= 2) {
+                        const pquote = chart[whichSet][l - 1];
+                        if (pquote) prevClose = pquote[field];
+                    }
+                    if (currentClose < prevClose) {
+                        backgroundColor = this.canvasStyle('stx_current_hr_down').backgroundColor;
+                        color = this.canvasStyle('stx_current_hr_down').color;
+                    } else {
+                        backgroundColor = this.canvasStyle('stx_current_hr_up').backgroundColor;
+                        color = this.canvasStyle('stx_current_hr_up').color;
+                    }
+                    if (chart.transformFunc) currentClose = chart.transformFunc(this, chart, currentClose);
+                    let txt;
+                    // If a chart panel, then always display at least the number of decimal places as calculated by masterData (panel.chart.decimalPlaces)
+                    // but if we are zoomed to high granularity then expand all the way out to the y-axis significant digits (panel.yAxis.printDecimalPlaces)
+                    let labelDecimalPlaces = Math.max(panel.yAxis.printDecimalPlaces, panel.chart.decimalPlaces);
+                    // ... and never display more decimal places than the symbol is supposed to be quoting at
+                    if (yAxis.maxDecimalPlaces || yAxis.maxDecimalPlaces === 0) labelDecimalPlaces = Math.min(labelDecimalPlaces, yAxis.maxDecimalPlaces);
+                    if (yAxis.priceFormatter) {
+                        txt = yAxis.priceFormatter(this, panel, currentClose, labelDecimalPlaces);
+                    } else {
+                        txt = this.formatYAxisPrice(currentClose, panel, labelDecimalPlaces);
+                    }
+                    const currentQuote = this.currentQuote();
+                    const y = this.pixelFromTransformedValue(currentClose, panel);
+                    const x = this.pixelFromTick(currentQuote.tick, chart) + (chart.lastTickOffset || 0);
+                    this.createYAxisLabel(panel, txt, y, backgroundColor, color);
+
+                    if (this.preferences.currentPriceLine === true && this.isHome()) {
+                        panel.chart.context.globalCompositeOperation = 'destination-over';
+                        this.plotLine(x, panel.right - 16, y, y, backgroundColor, 'segment', panel.chart.context, panel, {
+                            pattern: 'dashed',
```

Comment:
here i need to change the `x0` for plotLine to draw the line from current spot.
https://documentation.chartiq.com/CIQ.ChartEngine.html#plotLine